### PR TITLE
Fractal App 0.1.0

### PR DESCRIPTION
**Pull request by danigm**

### CODE REVIEW

File: org.gnome.Fractal.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "sdk": "org.gnome.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+    "command": "fractal",
+    "finish-args": ["--share=network", "--socket=x11", "--filesystem=host" ],
+    "build-options" : {
+        "env": {
+            "CARGO_HOME": "/run/build/Fractal/cargo",
+            "V": "1"
+        },
+        "build-args": [ "--share=network" ]
```
Comment (Barthalion) [inline org.gnome.Fractal.json]: You will have to figure out how to build it without network access.

File: org.gnome.Fractal.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+    "app-id": "org.gnome.Fractal",
+    "version": "master",
```
Comment (Barthalion) [inline org.gnome.Fractal.json]: Please remove it as it has no effect on the resulting flatpak; it's better to put version information into appdata.

File: org.gnome.Fractal.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            "V": "1"
+        },
+        "build-args": [ "--share=network" ]
+    },
+    "modules": [
+        {
+           "name": "Fractal",
+           "buildsystem": "simple",
+           "build-commands": [
+               "source /usr/lib/sdk/rust-stable/enable.sh && ./configure --prefix=/app && make && make install"
```
Comment (Barthalion) [inline org.gnome.Fractal.json]: I don't like it but I don't think there's any other way to source a file before build system kicks in. @TingPing @alexlarsson ideas?

File: org.gnome.Fractal.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+    "app-id": "org.gnome.Fractal",
+    "version": "master",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+    "command": "fractal",
+    "finish-args": ["--share=network", "--socket=x11", "--filesystem=host" ],
```
Comment (Barthalion) [inline org.gnome.Fractal.json]: Does it need `host` level access to the filesystem?

File: org.gnome.Fractal.json (hunk 3 above)
Comment (danigm) [reply org.gnome.Fractal.json]: The enable.sh file only adds rust to the PATH, but currently there's not way to expand environment vars. Once this issue https://github.com/flatpak/flatpak-builder/issues/18 is fixed we can remove it and use env build option:

**Comment by Barthalion** [comment]
bot, build org.gnome.Fractal

File: org.gnome.Fractal.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "app-id": "org.gnome.Fractal",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+    "command": "fractal",
+    "finish-args": [
+        "--share=network",
+        "--socket=x11",
```
Comment (TingPing) [inline org.gnome.Fractal.json]: `wayland` and `ipc` too.

File: org.gnome.Fractal.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    },
+    "modules": [
+        {
+           "name": "Fractal",
+           "buildsystem": "simple",
+           "build-commands": [
+               "source /usr/lib/sdk/rust-stable/enable.sh && ./configure --prefix=/app && make && make install"
+            ],
+           "sources": [
```
Comment (TingPing) [inline org.gnome.Fractal.json]: Forgot to install the appdata file, you can do it from `post-install`.

Comment (danigm) [reply org.gnome.Fractal.json]: The appdata file now is installed by the app make install so I've removed from this branch.

**Comment by Barthalion** [comment]
bot, build org.gnome.Fractal

File: org.gnome.Fractal.json (hunk 3 above)
Comment (alexlarsson) [reply org.gnome.Fractal.json]: You can use this:
https://github.com/flathub/org.gnome.gbrainy/blob/master/org.gnome.gbrainy.json#L21

**Comment by alexlarsson** [comment]
Merged

**Comment by nedrichards** [comment]
https://github.com/flathub/org.gnome.Fractal

**Comment by alexlarsson** [comment]
Hmm, i ran this and only get:

```
Query error: GuestLoginError(MatrixError(Object({"errcode": String("M_BAD_JSON"), "error": String("Content must be a JSON object.")})))
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
SYNC Error
...
```

**Comment by danigm** [comment]
Can you click in the top left button and login with your matrix user and password? The guest account also should work, the register part is not finished yet and doesn't work correctly.

**Comment by alexlarsson** [comment]
I get the same when i try to log in as guest too

**Comment by danigm** [comment]
It seems that in the 0.1.2 version doesn't work the guest login because we're sending an empty body and the server expect a json object... I've fixed that in the current git version. Anyway, the normal login by username and password should work.

**Comment by alexlarsson** [comment]
Ah, i finally was able to log in using the username "@gnomealex:matrix.org", where just "@gnomealex" or my email didn't work. Maybe this needs to be more accepting and/or better described.

**Pull request closed without merging**